Secp256k1.Net/LoadLibNative.cs

```diff
@@ -131,5 +131,32 @@ public static TDelegate GetDelegate<TDelegate>(IntPtr libPtr, string symbolName)
 
             return Marshal.GetDelegateForFunctionPointer<TDelegate>(functionPtr);
         }
+        public static TDelegate GetDelegateExternConst<TDelegate>(IntPtr libPtr, string symbolName)
```

Comment:
How is this function different from the existing `GetDelegate` function? Can you reuse the existing one?

Reply:
@zone117x 
It is to cater to the function ptr. https://github.com/zone117x/secp256k1/blob/8f24760236f7db30a2a3bd8340a92ce231947810/include/secp256k1.h#L561

Created an overloaded version of `GetDelegate` to explicitly show the difference.

```
public static TDelegate GetDelegate<TDelegate>(IntPtr libPtr, string symbolName, Func<IntPtr, IntPtr> pointerDereferenceFunc)
{
    var functionPtr = GetSymbolPointer(libPtr, symbolName);
    if (pointerDereferenceFunc != null)
    {
        functionPtr = pointerDereferenceFunc(functionPtr);
    }
    return Marshal.GetDelegateForFunctionPointer<TDelegate>(functionPtr);
}
```

```
static Lazy<TDelegate> LazyDelegate<TDelegate>(string symbol, Func<IntPtr, IntPtr> pointerDereferenceFunc)
{
    return new Lazy<TDelegate>(() =>
    {
        return LoadLibNative.GetDelegate<TDelegate>(_libPtr.Value, symbol, pointerDereferenceFunc);
    }, isThreadSafe: false);
}
```

And the function is used in this way where we use `Marshal.ReadIntPtr` to dereference the pointer.
```
static readonly Lazy<secp256k1_nonce_function> secp256k1_nonce_function_rfc6979
    = LazyDelegate<secp256k1_nonce_function>(nameof(secp256k1_nonce_function_rfc6979), Marshal.ReadIntPtr);
```